polling_stations/apps/data_importers/data_quality_report.py

```diff
@@ -309,224 +284,210 @@ class DataQualityReportBuilder:
-            self.report.append(
-                ANSI.bold(f"UPRNS ASSIGNED STATION ID        : {station_ids}")
+            table.add_row(
+                "UPRNS ASSIGNED STATION ID",
+                str(station_ids),
+                style="bold",
+                end_section=True,
             )
 
-            self.report.append(
-                ANSI.ok_bold(
-                    f" - As % of uprns in addressbase  : {round(100 * station_ids / uprns_in_council_area, 1)}%"
-                )
+            table.add_row(
+                " - As % of uprns in addressbase",
+                f"{round(100 * station_ids / uprns_in_council_area, 1)}%",
             )
             if self.csv_rows:
-                self.report.append(
-                    ANSI.ok_bold(
-                        f" - As % of rows in council csv   : {round(100 * station_ids / self.csv_rows, 1)}%"
-                    )
-                )
-            self.report.append("----------------------------------")
-            self.report.append(
-                ANSI.ok(
-                    "   - valid station id refs       : %i"
-                    % (address_report.get_addresses_with_valid_station_id_ref()),
+                table.add_row(
+                    " - As % of rows in council csv",
+                    f"{round(100 * station_ids / self.csv_rows, 1)}%",
                 )
+            table.add_row(
+                " - valid station id refs",
+                str(address_report.get_addresses_with_valid_station_id_ref()),
+                style="green",
             )
-            self.report.append(
-                ANSI.warning(
-                    "   - invalid station id refs     : %i"
-                    % (address_report.get_addresses_with_invalid_station_id_ref()),
-                )
+            table.add_row(
+                " - invalid station id refs",
+                str(address_report.get_addresses_with_invalid_station_id_ref()),
+                style="yellow",
             )
         else:
-
-            self.report.append(ANSI.warning("NO POLLING STATIONS ASSIGNED...      "))
-        self.report.append("----------------------------------")
-        self.report.append("\n")
+            table.add_row(
+                "NO POLLING STATIONS ASSIGNED", "", style="yellow", end_section=True
+            )
+        return table
 
     def build_report(self):
-        self.build_header()
-        self.build_district_report()
-        self.build_station_report()
-        self.build_address_report()
+        self.report.add_row(Panel(Text("DATA QUALITY REPORT", justify="center")))
+        if self.expecting_districts:
+            self.report.add_row(self.build_district_report())
+        self.report.add_row(self.build_station_report())
+        self.report.add_row(self.build_address_report())
 
     def output_console_report(self):
-        print("\n".join(self.report))
+        console = Console()
+        console.print(self.report)
```

Comment:
At the moment, this is the only place I (we) are using Rich so I've just instantiated a local ad-hoc `Console()` to render the report. If you start using Rich more widely in the application, you might want to switch to bootstrapping a global console object as described in https://rich.readthedocs.io/en/latest/console.html Tbh, given how much time you spend interacting with this application via the terminal, there may be some mileage in doing this.

Reply:
That's a good shout. It might be possible to bin https://github.com/DemocracyClub/UK-Polling-Stations/blob/master/polling_stations/apps/data_importers/loghelper.py and possibly produce output that is nicer and/or easier to work with